test/sharness/t0260-sharding.sh

```diff
@@ -8,22 +8,35 @@ test_description="Test directory sharding"
 
 . lib/test-lib.sh
 
+# We shard based on size with a threshold of 256 KiB (see config file docs)
+# above which directories are sharded.
+#
+# The directory size is estimated as the size of each link. Links are roughly
+# the entry name + the CID byte length (e.g. 34 bytes for a CIDv0). So for
+# entries of length 10 we need 256 KiB / (34 + 10) ~ 6000 entries in the
+# directory to trigger sharding.
 test_expect_success "set up test data" '
   mkdir testdata
   for i in `seq 2000`
   do
     echo $i > testdata/file$i
   done
+
+  mkdir big_dir
+  for i in `seq 5960` # Just above the number of entries that trigger sharding for 256KiB
+  do
+    echo $i > big_dir/`printf "file%06d" $i` # fixed length of 10 chars
+  done
```

Comment:
The previous `big-dir`/`small-dir` distinction might not have been the most appropriate for these two directories but I'm also not finding a clear distinction between `testdata` and `big-dir`, for example: there is little in `test_add_dir` that tells me which directory I'm working with (now that it has been removed from its arguments) and I need to parse through the code to find it's indeed `testdata`. (If we're keeping the `testdata` name for some kind of background compatibility ignore this comment.)

Reply:
ya, I mostly left it to keep the diff smaller here. These functions could reasonably be rewritten to take the input, output, and debugging directories as arguments, but we don't have to.

Reply:
Fair enough, prioritizing a small diff seems sensible with the amount of changes we're plumbing already.